lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3299,13 +3299,59 @@ private function dispatchOnFlushEvent()
         if ($this->evm->hasListeners(Events::onFlush)) {
             $this->evm->dispatchEvent(Events::onFlush, new OnFlushEventArgs($this->em));
         }
+        $subscribedSystems = [];
+        foreach ($this->entityChangeSets as $oid => $entityChangeSet) {
```

Comment:
@Ocramius this nested `foreach` would have a (small) performance impact, should we mind that?